src/qt3utils/applications/qt3ple/main.py

```diff
@@ -0,0 +1,395 @@
+import argparse
+import tkinter as tk
+import logging
+from threading import Thread
+import yaml
+import importlib
+import importlib.resources
+from typing import Any, Protocol
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
+import matplotlib
+import h5py
+import nidaqmx
+from qt3utils.nidaq.customcontrollers import VControl
+from qt3utils.datagenerators import plescanner
+from qt3utils.applications.controllers.nidaqedgecounter import QT3ScopeNIDAQEdgeCounterController
+
+matplotlib.use('Agg')
+
+parser = argparse.ArgumentParser(description='NI DAQ (PCIx 6363) / PLE Scanner',
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+parser.add_argument('-q', '--quiet', action='store_true',
+                    help='When true,logger level will be set to warning. Otherwise, set to "info".')
+parser.add_argument('-cmap', metavar='<MPL color>', default='gray',
+                    help='Set the MatplotLib colormap scale')
+
+args = parser.parse_args()
+
+logger = logging.getLogger(__name__)
+logging.basicConfig()
+
+if args.quiet is False:
+    logger.setLevel(logging.INFO)
+
+NIDAQ_DEVICE_NAMES = ['NIDAQ Rate Counter', "Lockin & Wavemeter"]
+RANDOM_DAQ_DEVICE_NAME = 'Random Data Generator'
+
+DEFAULT_DAQ_DEVICE_NAME = NIDAQ_DEVICE_NAMES[0]
+
+CONTROLLER_PATH = 'qt3utils.applications.controllers'
+STANDARD_CONTROLLERS = {NIDAQ_DEVICE_NAMES[0] : 'nidaq_rate_counter.yaml',
+                        NIDAQ_DEVICE_NAMES[1] : 'nidaq_wm_ple.yaml'}
+
+class ScanImage:
+    def __init__(self, mplcolormap='gray') -> None:
+        self.fig, self.ax = plt.subplots()
+        self.cbar = None
+        self.cmap = mplcolormap
+        self.cid = self.fig.canvas.mpl_connect('button_press_event', self.onclick)
+        self.ax.set_xlabel('Voltage')
+        self.log_data = False
+
+    def update(self, model) -> None:
+
+        if self.log_data:
+            data = np.log10(model.scanned_count_rate)
+            data[np.isinf(data)] = 0  # protect against +-inf
```

Comment:
Wouldn't it be better to set them to np.nan?

Reply:
Not in this instance, because it is specifically for outputting the graphical image to the GUI. I am pretty sure it was written this way so it doesn't error out or give unexpected behavior trying to render a nan. 
Also note that log(x)=+-inf only at x=0 for measurable values, so this logic tracks in my head.

Reply:
With matplotlib you can choose the `np.nan` behavior. Setting to the inf to zero can become a problem if you are gonna be plotting some form of normalized data. We could consider making it so that `np.nan` is printed as pink or another stark color, so that the user knows when its infinite. However, if this `data[np.isinf(data)] = 0` is consistent throughout the rest of qt3utils, we should leave it as is!